File: CPP/LeetCode/longestCommonPrefix.cpp
/*
 * Write a function to find the longest common prefix string amongst an array of strings.
 */
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string> &strs) {
        string result;
        if (strs.size() == 0) return result;
        for (int i = 0; i <= strs[0].size(); i++){
          char c = strs[0][i];
          for (int j = 0; j <= strs[j].size(); j++){
            if (i >= strs[j].size()) return result;
            if (strs[j][i] != c) return result;
            result += c;
          }
        }
        return result;
    }
};

int main(){
  vector<string> strs;
  strs.push_back("a");
  Solution *so = new Solution;
  cout << so->longestCommonPrefix(strs);
 }

File: LeetCode/longestCommonPrefix.py
"""
Write a function to find the longest common prefix string amongst an array of strings.
"""

class Solution:
    # @return a string
    def longestCommonPrefix(self, strs):
        result = ""
        if len(strs) == 0:
            return result
        for i in range(len(strs[0])):
            char = strs[0][i]
            try:
                for j in range(len(strs)):
                    if strs[j][i] != char:
                        return result
            except:
                return result
            result += char
        return result
